addons/website_version/models/ir_ui_view.py

```diff
@@ -0,0 +1,146 @@
+
+    _inherit = "ir.ui.view"
+
+    version_id = fields.Many2one('website_version.version', ondelete='cascade', string="version_id")
+
+    @api.multi
+    def write(self, vals, toggle=False):
+        if self.env.context is None:
+            self.env.context = {}
+
+        version_id = self.env.context.get('version_id')
+        #toggle is true when changing active field
+        if version_id and not self.env.context.get('write_on_view') and not toggle:
+            self.env.context = dict(self.env.context, write_on_view=True)
+            snap = self.env['website_version.version']
+            version = snap.browse(version_id)
+            website_id = version.website_id.id
+            version_view_ids = self.env['ir.ui.view']
+            for current in self:
+                #check if current is in version
+                if current.version_id.id == version_id:
+                    version_view_ids += current
+                else:
+                    new_v = self.search([('website_id', '=', website_id), ('version_id', '=', version_id), ('key', '=', current.key)])
+                    if new_v:
+                        version_view_ids += new_v
+                    else:
+                        copy_v = current.copy({'version_id': version_id, 'website_id': website_id})
+                        version_view_ids += copy_v
+            super(view, version_view_ids).write(vals)
+        else:
+            self.env.context = dict(self.env.context, write_on_view=True)
+            super(view, self).write(vals)
+
+    @api.one
+    def publish(self):
+        #To delete and replace views which are in the website( in fact with website_id)
+        master_record = self.search([('key', '=', self.key), ('version_id', '=', False), ('website_id', '=', self.website_id.id)])
+        if master_record:
+            master_record.unlink()
+        self.copy({'version_id': None})
+
+    #To publish a view in backend
+    @api.model
+    def action_publish(self, ids):
+        if self.env.context is None:
+            self.env.context = {}
+        view_id = self.env.context.get('active_id')
+        self.browse(view_id).publish()
+
+    @api.model
+    def get_view_id(self, xml_id):
+        if self.env.context and 'website_id' in self.env.context and not isinstance(xml_id, (int, long)):
+            domain = [('key', '=', xml_id), '|', ('website_id', '=', self.env.context['website_id']), ('website_id', '=', False)]
+            if 'version_id' in self.env.context:
+                domain += ['|', ('version_id', '=', self.env.context['version_id']), ('version_id', '=', False)]
+            xml_id = self.search(domain, order='website_id,version_id', limit=1).id
+        else:
+            xml_id = super(view, self).get_view_id(xml_id)
+        return xml_id
+
+    _read_template_cache = dict(accepted_keys=('lang', 'inherit_branding', 'editable', 'translatable', 'website_id', 'version_id'))
+
+    @tools.ormcache_context(**_read_template_cache)
```

Comment:
Why not directly accepted_keys=(...) ? your variable does not seem to be re-used somewhere.